src/libraries/Microsoft.PowerFx.Core/Public/Engine.cs

```diff
@@ -31,27 +35,87 @@ public class Engine : IPowerFxEngine
         /// <param name="powerFxConfig"></param>
         public Engine(PowerFxConfig powerFxConfig)
         {
-            powerFxConfig.Lock();
             Config = powerFxConfig;
         }
 
+        // All functions that powerfx core knows about. 
+        // Derived engines may only support a subset of these builtins, 
+        // and they may add their own custom ones. 
+        private static readonly ReadOnlySymbolTable _allBuiltinCoreFunctions = NewDefault(Core.Texl.BuiltinFunctionsCore.BuiltinFunctionsLibrary);
+
+        /// <summary>
+        /// Builtin functions supported by this engine. 
+        /// </summary>
+        public ReadOnlySymbolTable SupportedFunctions { get; set; } = _allBuiltinCoreFunctions;
+
+        // Helper to creaet a ReadOnly symbol table around a set of core functions.
+        // Important that this is readonly so that it can be safely shared across engines. 
+        internal static ReadOnlySymbolTable NewDefault(IEnumerable<TexlFunction> coreFunctions)
```

Comment:
>NewDefault [](http://example.com/codeflow?start=44&length=10)

organization nit: this builder static seems to logically belong on ReadOnlySymbolTable, not Engine.